From 9f75586fd95f7e965dbc76bfa3b541ddc3b92728 Mon Sep 17 00:00:00 2001
From: Valentin Caron <valentin.caron@foss.st.com>
Date: Thu, 24 Apr 2025 17:52:02 +0200
Subject: [PATCH 1691/1695] dts: stm32: change scmi for CM33 shared memory
 address on stm32mp25 soc

Use a dedicated memory region for the shared memory used for SCMI commands
between CA35_S an CM33_NS.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git bc556d5ab3f39dcd2b837c08756d5e7471ab06b8 dts: stm32: change scmi for CM33 shared memory address on stm32mp25 soc]
Signed-off-by: Valentin Caron <valentin.caron@foss.st.com>
Change-Id: I6d075906c8745948fa8ec554a6476ea260139ad1
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/453944
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Reviewed-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
---
 core/arch/arm/dts/stm32mp25-st-scmi-cfg.dtsi | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/arch/arm/dts/stm32mp25-st-scmi-cfg.dtsi b/core/arch/arm/dts/stm32mp25-st-scmi-cfg.dtsi
index edd6bb632..ac79926d0 100644
--- a/core/arch/arm/dts/stm32mp25-st-scmi-cfg.dtsi
+++ b/core/arch/arm/dts/stm32mp25-st-scmi-cfg.dtsi
@@ -950,7 +950,7 @@
 			scmi-channel-id = <0>;
 			mbox-names = "scmi_cm33";
 			mboxes = <&ipcc1 RIF_IPCC_CPU1_CHANNEL(16)>;
-			shmem = <&ipc_shmem>;
+			shmem = <&scmi_cid2_ns>;
 			#address-cells = <1>;
 			#size-cells = <0>;
 
-- 
2.47.2

